packages/block-library/src/latest-posts/style.scss

```diff
@@ -40,3 +44,22 @@
 	margin-top: $grid-size;
 	margin-bottom: $grid-size-large;
 }
+
+.wp-block-latest-posts__featured-image {
+	img {
+		height: auto;
+		width: auto;
+	}
+	&.alignleft {
```

Comment:
The `alignleft` and `alignright` classes are intended to be used with CSS float styles (and they probably shouldn't be called `alignleft` and `alignright`, but that's a whole other topic). This class style override is inconsistent with the usage of the class name elsewhere.